### PR TITLE
Add YAML & TOML support, check for duplicates & add tests

### PR DESCRIPTION
**Pull request by S1SYPHOS**
Hey there,
as you can easily see, I took a different path with my implementation. While providing support for both YAML and TOML, this PR also does some refactoring, checks for duplicates (as discussed in #14) and updates the README accordingly.

In opposition to the (AFAIK conflicting) PR from @wessven in #13, I decided to let a class handle all tasks related to parsing data files. Since the PR has been open and unresolved for quite some time, I decided to start from scratch. Furthermore, `tomli` is now only installed if `python_version < "3.11"`. Also if `tomli`/`tomllib` or `pyyaml` are not available, their respective data files are ignored while issuing an `INFO` log entry. Each data source handler has its own parsing function, making it very easy to add new filetypes in the future in the same manner. The `Parser` class now single-handedly decides which extensions are supported, how to open/process any given file.

Lastly, I took the liberty of extending docblocks, comments, typehinting and correcting minor typos :stuck_out_tongue: 

Cheers 
S1SYPHOS

### CODE REVIEW

**Comment by S1SYPHOS** [comment]
What about adding tests? This should prove useful in the long run - I'll see if I can whip something up sometime soon 🤣

**Comment by S1SYPHOS** [comment]
Since the `DataParser` class does not depend on `pelican`, a simple `pytest` setup should suffice for now ..

**Comment by LucasVanHaaren** [comment]
Hey @S1SYPHOS,

Thank you for your work, I will review and test it as soon as possible !

@wessven, as you have worked on this same feature, can you share me what do you think of this implementation ?

**Comment by S1SYPHOS** [comment]
Thank you, kind Sir!

File: README.md [hunk 1]
```diff
@@ -89,7 +104,7 @@ Users can easily fetch the files in their pelican project by running the followi
 pelican-data-files --fetch <THEME_NAME>
 ```
 
-This will copy the `data/` folder of the theme, into the `data/` folder of their Pelican project.
+This will copy the `data/` folder of the theme into the `data/` folder of their Pelican project.
```
Comment (LucasVanHaaren) [inline README.md]: Assuming your modifications, now themes files will be copied at `content/data/` instead of `data/`

**Review by LucasVanHaaren — COMMENTED** [review]
Great, works like a charm !

I encountered an issue about the dependency markupSafe (pulled by pelican dep) which do breaking change to pelican. Have you got [this issue](https://github.com/getpelican/pelican/issues/2983) ?

I was thinking about setting a fixed version of markupsafe to avoid future issues, what do you think about this ?

PS: When installing project dependencies I got this versions:

```yaml
[...]
Jinja2==3.1.2
MarkupSafe==2.1.1
pelican==4.8.0
[...]
```
Using markupsafe<2.10 fix the issue

**Comment by S1SYPHOS** [comment]
It's a known issue, I encountered this, too 😕 pinning the Version seems reasonable (until it's fixed)

**Comment by LucasVanHaaren** [comment]
Okay great, so we will set to a fixed the version in order to avoid this issue !

Can you apply the following changes ?

- edit in `README.md` the destination themes data-files path from `data/` to `content/data/`
- setting dependency version in `setup.cfg` to `markupsafe<2.10`

Then I will merge your awesome PR :sunglasses: :partying_face:

**Comment by S1SYPHOS** [comment]
done :sunglasses:

**Review by LucasVanHaaren — APPROVED** [review]

**Comment by LucasVanHaaren** [comment]
Thanks a lot for your work, I just merged your commits into main :smiley: 
I created a release tagged as 1.2.0, I will also publish this version to pypi ASAP

**Comment by wessven** [comment]
@S1SYPHOS, thanks for your work on this!

@LucasVanHaaren, apologies for not getting back to you.  I was going through a crazy time when @S1SYPHOS was doing his work.  I didn't mind which implementation ended up getting used, as long as TOML support could be made available for Pelican.

Just out of curiosity, what was wrong with my PR #13 that it initially didn't get merged?

**Comment by S1SYPHOS** [comment]
Thanks for your kind words, hope stuff is looking up for you!

**Comment by LucasVanHaaren** [comment]
@wessven There was no problem with your implementation of the TOML support feature. However, when you sent your PR, I was totally running out of time. In addition I had a lot of issues while setting up and testing your PR because it had been a while since I had not coded with this environment. I didn’t want to merge the code right after the review, without testing it on my side.

I want to thank you for your work on this feature, even if it is not this one that was merged :smiley:

**Pull request merged**